### PR TITLE
leave out templates for training

### PR DESCRIPTION
**Pull request by Angryrou**
1. fixed a bug during the joint training - the previous val loss was computed incorrectly and can not be reproduced for the final ckpt weights. (run `pip install -r requirements.txt` ahead)
2. add data split to support the out-of-distribution test set with 10 different folds.

### CODE REVIEW

**Comment by Angryrou** [comment]
* **#73** <a href="https://app.graphite.dev/github/pr/Angryrou/udao-spark-optimizer/73?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Angryrou and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>
